### ISSUE TITLE
Dispatch cached API payloads

### ISSUE THREAD

**Issue by alanfriedman** [agraboso/redux-api-middleware#173]
Hi! Thanks for the great work on this.

I'm looking for a way to dispatch the success action (and apply any `payload` and `meta` transformations) but using a cached API response (from local storage, Redis, etc) instead of the real one. In this case, the actual fetch would be skipped.

The alternative to doing this is manually dispatching the success action, but then you need to also apply the various transformations, check if `type` is a string or object, etc - basically rewrite some functionality of the library itself.

I'm happy to put in a PR if this seems valuable. Or if there's a good way to do this that I'm missing, let me know. Thanks!

**Comment by nason** [comment]
👋Alan, hope all is well!

Yes, this would be a great enhancement. The alternative you describe can achieve this effect, but you're right, this gets complex very quickly.

I think that tying this library to a particular caching mechanism would not be a great direction. A thought thats come up a few times for enabling this, and other types of custom behavior, is supporting an optional `fetch` property on the `RSAA` to pass in an a Fetch API abiding wrapper that houses your custom logic (theres lots of examples out there already, like   https://www.npmjs.com/package/make-fetch-happen):

```diff
{
  [RSAA]: {
    endpoint: 'http://www.example.com/api/users',
    method: 'GET',
    types: ['REQUEST', 'SUCCESS', 'FAILURE']
+   fetch: myCustomFetchWrapper
  }
}
```

What do you think? Would that meet your needs? If you want to open a PR, that would be great! Otherwise, I think I should be able to work on something like this this in a week or so.

**Comment by alanfriedman** [comment]
Thanks for the quick response!

That makes sense to me, and would also allow customizing other things, like what constitutes a “successful” response.

So the custom fetch option could be a function that takes the endpoint and fetch options (headers, body, method, etc) as args and returns a promise that resolves with the payload, or rejects with an error. Is that what you had in mind?

**Comment by alanfriedman** [comment]
Nevermind, I see what you mean. They would return their own implementation of the fetch API. I should be able to get something started tomorrow as a first pass.

**Comment by alanfriedman** [comment]
Working on this in #174

**Comment by alanfriedman** [comment]
Fixed by #174
